File: 657 - Robot Return to Origin.java
class Solution {
    public boolean judgeCircle(String moves) {
        int u = 0, d = 0, l = 0, r = 0;
        for (char move : moves.toCharArray()) {
            switch (move) {
                case 'U': u++;
                break;
                case 'D': d++;
                break;
                case 'L': l++;
                break;
                case 'R': r++;
            }
        }
        return u == d && l == r;
    }
}

File: 657 - Robot Return to Origin.js
/**
 * @param {string} moves
 * @return {boolean}
 */
var judgeCircle = function (moves) {
    let l = /L/g;
    let r = /R/g;
    let u = /U/g;
    let d = /D/g;
    let cl = moves.match(l) == null ? 0 : moves.match(l).length;
    let cr = moves.match(r) == null ? 0 : moves.match(r).length;
    let cu = moves.match(u) == null ? 0 : moves.match(u).length;
    let cd = moves.match(d) == null ? 0 : moves.match(d).length;

    return cl == cr && cu == cd;
};